src/Options.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace LanguageServer;
+
+class Options
+{
+    /**
+     * Filetypes the indexer should process
+     *
+     * @var string[]
+     */
+    public $fileTypes = ['.php'];
+
+    /**
+     * List of options that affect the indexer
+     */
+    private $indexerOptions = ['fileTypes'];
+
+    /**
+     * Validate/Filter input and set options for file types
+     *
+     * @param array $fileTypes List of file types
+     */
+    public function setFileTypes(array $fileTypes)
+    {
+        $fileTypes = filter_var_array($fileTypes, FILTER_SANITIZE_STRING);
+        $fileTypes =  filter_var($fileTypes, FILTER_CALLBACK, ['options' => [$this, 'filterFileTypes']]); // validate file type format
+        $fileTypes = array_filter($fileTypes, 'strlen'); // filter empty items
+        $fileTypes = array_values($fileTypes); //rebase indexes
+
+        $this->fileTypes = !empty($fileTypes) ? $fileTypes : $this->fileTypes;
+    }
+
+    /**
+     * Get list with options that affect the indexer
+     *
+     * @return array
+     */
+    public function getIndexerOptions(): array
```

Comment:
I would like to keep `Options` a dumb data object (like other protocol structures).
Checking if reindexing is needed should be done solely in `didChangeConfiguration`.